src/shared/components/oncoprint/ResultsViewOncoprintUtils.tsx

```diff
@@ -273,16 +276,29 @@ export function getGenericAssayTrackCacheQueries(
             );
             return _.keys(entry.entities).map(entityId => {
                 const entity = genericAssayEntitiesByEntityId[entityId];
-                const entityName = getGenericAssayMetaPropertyOrDefault(
+                let entityName = getGenericAssayMetaPropertyOrDefault(
                     entity,
                     COMMON_GENERIC_ASSAY_PROPERTY.NAME,
                     entityId
                 );
-                const description = getGenericAssayMetaPropertyOrDefault(
+                let description = getGenericAssayMetaPropertyOrDefault(
                     entity,
                     COMMON_GENERIC_ASSAY_PROPERTY.DESCRIPTION,
                     entityName
                 );
+                // Override name and description based on GenericAssayConfig
+                entityName = GENERIC_ASSAY_CONFIG.genericAssayConfigByType[type]
```

Comment:
kinda confusing.   maybe just use the ternary above in the original assignment rather than assign and then override that assignement here